### PR TITLE
663-AccessTokens validate expiration date

### PR DESCRIPTION
**Pull request by cd-bitwarden**
## Type of change

```
- [ x ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Validate expiration date is in the future on the server side

## Code changes

src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs - implement IValidatableObject so that if the expiration date is in the past it returns an error message that's descriptive
## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs [hunk 1]
```diff
@@ -34,4 +34,13 @@ public ApiKey ToApiKey(Guid serviceAccountId)
             EncryptedPayload = EncryptedPayload,
         };
     }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        if (ExpireAt <= DateTime.Now)
```
Comment (Thomas-Avery) [inline src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs:40]:
```suggestion
        if (ExpireAt <= DateTime.UtcNow)
```

We send and handle time as UTC.

I would also add some logic here to handle the null case for when we are doing never expiring tokens.

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
LGTM

File: src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs [hunk 2]
```diff
@@ -34,4 +34,13 @@ public ApiKey ToApiKey(Guid serviceAccountId)
             EncryptedPayload = EncryptedPayload,
         };
     }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        if (ExpireAt <= DateTime.UtcNow && ExpireAt != null)
```
Comment (Thomas-Avery) [inline src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs:40]:
```suggestion
        if (ExpireAt != null && ExpireAt <= DateTime.UtcNow)
```

The conditional will short circuit if the left most check is false. So we would want our null check first.

Here are some [docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-and-operator-) that will explain it better.

Comment (cd-bitwarden) [reply src/Api/SecretsManager/Models/Request/AccessTokenCreateRequestModel.cs]: Thank you Thomas!

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Pull request merged**